heartbeat/monitors/active/http/checkjson.go

```diff
@@ -0,0 +1,175 @@
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/PaesslerAG/gval"
+	"github.com/PaesslerAG/jsonpath"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/v7/libbeat/common/jsontransform"
+	"github.com/elastic/beats/v7/libbeat/conditions"
+)
+
+type jsonChecker func(interface{}) bool
+type compiledJsonCheck struct {
+	description string
+	check       jsonChecker
+	source      string
+}
+
+func checkJson(checks []*jsonResponseCheck) (bodyValidator, error) {
+	var expressionChecks []compiledJsonCheck
+	var conditionChecks []compiledJsonCheck
+
+	for _, check := range checks {
+		if check.Expression != "" && check.Condition != nil {
+			return nil, fmt.Errorf("only one of 'expression' and 'condition' can be specified for JSON check '%s'", check.Description)
+		}
+
+		if check.Expression != "" {
+			eval, err := gval.Full(jsonpath.PlaceholderExtension()).NewEvaluable(check.Expression)
+			if err != nil {
+				return nil, fmt.Errorf("could not compile gval expression '%s': %w", check.Expression, err)
+			}
+
+			checkFn := func(d interface{}) bool {
+				matches, err := eval.EvalBool(context.Background(), d)
+				if err != nil {
+					// Conditions cannot match array root JSON responses
+					return false
+				}
+
+				return matches
+			}
+
+			expressionChecks = append(expressionChecks, compiledJsonCheck{
+				description: check.Description,
+				check:       checkFn,
+				source:      check.Expression,
+			})
+		} else if check.Condition != nil {
+			cfgwarn.Deprecate("8.0.0", "JSON conditions are deprecated, use 'expression' instead.")
+			cond, err := conditions.NewCondition(check.Condition)
+			if err != nil {
+				return nil, fmt.Errorf("could not load JSON condition '%s': %w", check.Description, err)
+			}
+
+			checkFn := func(d interface{}) bool {
+				ms, ok := d.(map[string]interface{})
+				if ok {
+					return cond.Check(common.MapStr(ms))
+				} else {
+					return false
+				}
+			}
+			conditionChecks = append(conditionChecks, compiledJsonCheck{
+				description: check.Description,
+				check:       checkFn,
+				source:      fmt.Sprintf("%v", check.Condition),
```

Comment:
Any reason for not using check.Condition as it is?

Reply:
Yes, there's no `String()` method on it. `%v` yields something somewhat debuggable in the event of an error. I'd say we should do more here, but since we're deprecating it, I think this is fine.